### PR TITLE
Added role so that the non-admin will not be able to unlink the board

### PR DESCRIPTION
**Pull request by Rajat-Dabade**
#### Summary
This PR will add functionality to not allow non-admin to unlink the board from rhs-channel

#### Ticket Link
Fixes https://github.com/mattermost/focalboard/issues/3600

### CODE REVIEW

**Review by sbishel — CHANGES_REQUESTED** [review]
Is there a reason you didn't use <BoardPermissionGate>? I think you simply need to wrap the <MenuWrapper> in a <BoardsPermissionGate> like was done here.
https://github.com/mattermost/focalboard/blob/main/webapp/src/components/shareBoard/userPermissionsRow.tsx#L57

Also we should avoid mixing Bug fixes with UI Enhancements.

**Comment by Rajat-Dabade** [comment]
Thanks for the feedback @sbishel, it was my bad I was not aware of the `invert` prop. I have made the changes and pushed it. And regarding the UI changes, It was needed for this PR and also confirmed it with @asaadmahmood. Should we go forward with it?

**Comment by asaadmahmood** [comment]
I think @sbishel is more referring to some other minor UI changes I did on the side.
Noted Scott. I also fixed some UI bugs as well in this, that may have taken more time to create an issue and a PR (like the linked boards in the channels RHS not being theme compatible, etc).

And some spacing/sizing issues.

When I work on a part of the code, I also see if the UI on that part can also be improved if its just a few lines of fix.

File: mattermost-plugin/webapp/src/components/rhsChannelBoardItem.tsx [hunk 1]
```diff
@@ -55,18 +58,41 @@ const RHSChannelBoardItem = (props: Props) => {
...
+                                id='unlinkBoard'
+                                disabled={true}
+                                name={intl.formatMessage({id: 'rhs-boards.unlink-board1', defaultMessage: 'Unlink board Hello'})}
+                                icon={<DeleteIcon/>}
+                                onClick={() => {
+                                    onUnlinkBoard(board)
+                                }}
+                                subText={intl.formatMessage({id: 'rhs-board-non-admin-msg', defaultMessage:'You are not an admin of the board'})}
+                            />
+                        </BoardPermissionGate>
```
Comment (sbishel) [inline mattermost-plugin/webapp/src/components/rhsChannelBoardItem.tsx:95]: This section should be deleted.

**Review by sbishel — CHANGES_REQUESTED** [review]
@rajat @asaadmahmood - Please revert all of the UX fixes in this PR. This PR should be essentially a 3 code line/3 file change. Instead there are 38 files that have changed. I would like the UX changes pulled out separately into another PR. 

I'm ok with doing minor UI fixes in with a PR that is implementing a new feature. But when it is a bug fix, I would like to restrict those to fixing the issue. When a "minor" UI fix causes 30+ snapshot files to change, then it should be in its own PR. These changes in this PR cause it go from a couple files being changed to 38 files. On this PR, I feel 5/5 on this.

**Comment by sbishel** [comment]
@Rajat-Dabade @asaadmahmood - Sorry, I didn't read what the entire fix entailed. I just expected the menu item to be hidden. Sorry for that confusion. I don't like the feature. We don't disable menus anywhere else that I know of or tell users they can't do something because they are not an admin. But ignore my comments for now. It looks good base on the instructions in the PR....sorry.

**Review by sbishel — APPROVED** [review]
Approving from a code perspective.

**Comment by sbishel** [comment]
@Rajat-Dabade - Ready to merge

**Comment by rajat** [comment]
Kindly, remove me from the github.
My github handle is rajat.
I am not part of this thread.


Thanks and Regards,
Rajat Kumar

**Pull request merged**